### ISSUE TITLE
The value of enums shoulde be inferred if possible

### ISSUE THREAD

**Issue by saryou** [microsoft/pyright#1624]
**Is your feature request related to a problem? Please describe.**

I know that enum's value is defined as Any in enum.pyi but it is better if the value type inferred precisely. Also mypy already complains that following code has an error.

```
from enum import Enum


class MyEnum(Enum):
    a = 1


def take_str(s: str):
    print(s)


def take_int(s: int):
    print(s)


take_str(MyEnum.a.value)
take_int(MyEnum.a.value)
```

**Describe the solution you'd like**

I don't understand the details but I think this is the mypy's implementation:

https://github.com/python/mypy/pull/6867

**Comment by erictraut** [comment]
Thanks for the suggestion. This support will be included in the next release of pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.121, which I just published. It will also be included in the next release of pylance.
